### PR TITLE
INTYGFV-15160: Implemented categories hypoglykemi, ovrigt and bedomning with questions.

### PR DESCRIPTION
**Pull request by mikeyOne**

### CODE REVIEW

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiAllvarligSenasteTolvManaderna.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        final var allvarligSenasteTolvManaderna =
+            hypoglykemi != null && hypoglykemi.getAllvarligSenasteTolvManaderna() != null
+                ? hypoglykemi.getAllvarligSenasteTolvManaderna() : null;
+        return CertificateDataElement.builder()
+            .id(HYPOGLYKEMI_ALLVARLIG_SENASTE_TOLV_MANADERNA_SVAR_ID)
+            .parent(HYPOGLYKEMI_CATEGORY_ID)
+            .index(index)
+            .config(
+                CertificateDataConfigRadioBoolean.builder()
+                    .id(HYPOGLYKEMI_ALLVARLIG_SENASTE_TOLV_MANADERNA_DELSVAR_ID)
```
Comment (ChrilleSigerhed) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiAllvarligSenasteTolvManaderna.java:56]: Should the config.id be HYPOGLYKEMI_ALLVARLIG_SENASTE_TOLV_MANADERNA_JSON_ID instead? This has no functional impact from what I can tell, but to stay uniformed in how we design our configs, consider making this change.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiAterkommandeSenasteAret.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    public static CertificateDataElement toCertificate(Hypoglykemi hypoglykemi, int index, CertificateTextProvider textProvider) {
+        final var aterkommandeSenasteAret =
+            hypoglykemi != null && hypoglykemi.getAterkommandeSenasteAret() != null ? hypoglykemi.getAterkommandeSenasteAret() : null;
+        return CertificateDataElement.builder()
+            .id(HYPOGLYKEMI_ATERKOMMANDE_SENASTE_ARET_SVAR_ID)
+            .parent(HYPOGLYKEMI_CATEGORY_ID)
+            .index(index)
+            .config(
+                CertificateDataConfigRadioBoolean.builder()
+                    .id(HYPOGLYKEMI_ATERKOMMANDE_SENASTE_ARET_DELSVAR_ID)
```
Comment (ChrilleSigerhed) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiAterkommandeSenasteAret.java:55]: Should the config.id be HYPOGLYKEMI_ATERKOMMANDE_SENASTE_ARET_JSON_ID instead? This has no functional impact from what I can tell, but to stay uniformed in how we design our configs, consider making this change.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiAterkommandeVaketSenasteTolv.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        final var aterkommandeVaketSenasteTolv =
+            hypoglykemi != null && hypoglykemi.getAterkommandeVaketSenasteTolv() != null
+                ? hypoglykemi.getAterkommandeVaketSenasteTolv() : null;
+        return CertificateDataElement.builder()
+            .id(HYPOGLYKEMI_ATERKOMMANDE_VAKET_SENASTE_TOLV_SVAR_ID)
+            .parent(HYPOGLYKEMI_CATEGORY_ID)
+            .index(index)
+            .config(
+                CertificateDataConfigRadioBoolean.builder()
+                    .id(HYPOGLYKEMI_ATERKOMMANDE_VAKET_SENASTE_TOLV_DELSVAR_ID)
```
Comment (ChrilleSigerhed) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiAterkommandeVaketSenasteTolv.java:55]: Should the config.id be HYPOGLYKEMI_ATERKOMMANDE_VAKET_SENASTE_TOLV_JSON_ID instead? This has no functional impact from what I can tell, but to stay uniformed in how we design our configs, consider making this change.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiFormagaKannaVarningstecken.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    public static CertificateDataElement toCertificate(Hypoglykemi hypoglykemi, int index, CertificateTextProvider textProvider) {
+        final var kannaVarningstecken =
+            hypoglykemi != null && hypoglykemi.getFormagaKannaVarningstecken() != null ? hypoglykemi.getFormagaKannaVarningstecken() : null;
+        return CertificateDataElement.builder()
+            .id(HYPOGLYKEMI_FORMAGA_KANNA_VARNINGSTECKEN_SVAR_ID)
+            .parent(HYPOGLYKEMI_CATEGORY_ID)
+            .index(index)
+            .config(
+                CertificateDataConfigRadioBoolean.builder()
+                    .id(HYPOGLYKEMI_FORMAGA_KANNA_VARNINGSTECKEN_DELSVAR_ID)
```
Comment (ChrilleSigerhed) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiFormagaKannaVarningstecken.java:55]: Should the config.id be HYPOGLYKEMI_FORMAGA_KANNA_VARNINGSTECKEN_JSON_ID instead? This has no functional impact from what I can tell, but to stay uniformed in how we design our configs, consider making this change.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiForstarRiskerMedHypoglykemi.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        final var forstarRisker =
+            hypoglykemi != null && hypoglykemi.getForstarRiskerMedHypoglykemi() != null
+                ? hypoglykemi.getForstarRiskerMedHypoglykemi() : null;
+        return CertificateDataElement.builder()
+            .id(HYPOGLYKEMI_FORSTAR_RISKER_MED_HYPOGLYKEMI_SVAR_ID)
+            .parent(HYPOGLYKEMI_CATEGORY_ID)
+            .index(index)
+            .config(
+                CertificateDataConfigRadioBoolean.builder()
+                    .id(HYPOGLYKEMI_FORSTAR_RISKER_MED_HYPOGLYKEMI_DELSVAR_ID)
```
Comment (ChrilleSigerhed) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiForstarRiskerMedHypoglykemi.java:56]: Should the config.id be HYPOGLYKEMI_FORSTAR_RISKER_MED_HYPOGLYKEMI_JSON_ID instead? This has no functional impact from what I can tell, but to stay uniformed in how we design our configs, consider making this change.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiKontrollSjukdomstillstand.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    public static CertificateDataElement toCertificate(Hypoglykemi hypoglykemi, int index, CertificateTextProvider textProvider) {
+        final var kontroll =
+            hypoglykemi != null && hypoglykemi.getKontrollSjukdomstillstand() != null ? hypoglykemi.getKontrollSjukdomstillstand() : null;
+        return CertificateDataElement.builder()
+            .id(HYPOGLYKEMI_KONTROLL_SJUKDOMSTILLSTAND_SVAR_ID)
+            .parent(HYPOGLYKEMI_CATEGORY_ID)
+            .index(index)
+            .config(
+                CertificateDataConfigRadioBoolean.builder()
+                    .id(HYPOGLYKEMI_KONTROLL_SJUKDOMSTILLSTAND_DELSVAR_ID)
```
Comment (ChrilleSigerhed) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiKontrollSjukdomstillstand.java:55]: Should the config.id be HYPOGLYKEMI_KONTROLL_SJUKDOMSTILLSTAND_JSON_ID instead? This has no functional impact from what I can tell, but to stay uniformed in how we design our configs, consider making this change.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiRegelbundnaBlodsockerkontroller.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        final var regelbundnaBlodsockerkontroller =
+            hypoglykemi != null && hypoglykemi.getRegelbundnaBlodsockerkontroller() != null
+                ? hypoglykemi.getRegelbundnaBlodsockerkontroller() : null;
+        return CertificateDataElement.builder()
+            .id(HYPOGLYKEMI_REGELBUNDNA_BLODSOCKERKONTROLLER_SVAR_ID)
+            .parent(HYPOGLYKEMI_CATEGORY_ID)
+            .index(index)
+            .config(
+                CertificateDataConfigRadioBoolean.builder()
+                    .id(HYPOGLYKEMI_REGELBUNDNA_BLODSOCKERKONTROLLER_DELSVAR_ID)
```
Comment (ChrilleSigerhed) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiRegelbundnaBlodsockerkontroller.java:57]: Should the config.id be HYPOGLYKEMI_REGELBUNDNA_BLODSOCKERKONTROLLER_JSON_ID instead? This has no functional impact from what I can tell, but to stay uniformed in how we design our configs, consider making this change.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiVidtaAdekvataAtgarder.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    public static CertificateDataElement toCertificate(Hypoglykemi hypoglykemi, int index, CertificateTextProvider textProvider) {
+        final var adekvataAtgarder =
+            hypoglykemi != null && hypoglykemi.getVidtaAdekvataAtgarder() != null ? hypoglykemi.getVidtaAdekvataAtgarder() : null;
+        return CertificateDataElement.builder()
+            .id(HYPOGLYKEMI_VIDTA_ADEKVATA_ATGARDER_SVAR_ID)
+            .parent(HYPOGLYKEMI_CATEGORY_ID)
+            .index(index)
+            .config(
+                CertificateDataConfigRadioBoolean.builder()
+                    .id(HYPOGLYKEMI_VIDTA_ADEKVATA_ATGARDER_DELSVAR_ID)
```
Comment (ChrilleSigerhed) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiVidtaAdekvataAtgarder.java:55]: Should the config.id be HYPOGLYKEMI_VIDTA_ADEKVATA_ATGARDER_JSON_ID instead? This has no functional impact from what I can tell, but to stay uniformed in how we design our configs, consider making this change.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionOvrigtBorUndersokasAvSpecialist.java [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    public static CertificateDataElement toCertificate(Ovrigt ovrigt, int index, CertificateTextProvider textProvider) {
+        final var borUndersokasAvSpecialist =
+            ovrigt != null && ovrigt.getBorUndersokasAvSpecialist() != null ? ovrigt.getBorUndersokasAvSpecialist() : null;
+        return CertificateDataElement.builder()
+            .id(OVRIGT_BOR_UNDERSOKAS_AV_SPECIALIST_SVAR_ID)
+            .parent(OVRIGT_CATEGORY_ID)
+            .index(index)
+            .config(
+                CertificateDataConfigTextArea.builder()
+                    .id(OVRIGT_BOR_UNDERSOKAS_AV_SPECIALIST_DELSVAR_ID)
```
Comment (ChrilleSigerhed) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionOvrigtBorUndersokasAvSpecialist.java:50]: Should the config.id be OVRIGT_BOR_UNDERSOKAS_AV_SPECIALIST_JSON_ID instead? This has no functional impact from what I can tell, but to stay uniformed in how we design our configs, consider making this change.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionOvrigtKomplikationerAvSjukdomen.java [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    public static CertificateDataElement toCertificate(Ovrigt ovrigt, int index, CertificateTextProvider textProvider) {
+        final var komplikationerAvSjukdomen =
+            ovrigt != null && ovrigt.getKomplikationerAvSjukdomen() != null ? ovrigt.getKomplikationerAvSjukdomen() : null;
+        return CertificateDataElement.builder()
+            .id(OVRIGT_KOMPLIKATIONER_AV_SJUKDOMEN_SVAR_ID)
+            .parent(OVRIGT_CATEGORY_ID)
+            .index(index)
+            .config(
+                CertificateDataConfigRadioBoolean.builder()
+                    .id(OVRIGT_KOMPLIKATIONER_AV_SJUKDOMEN_DELSVAR_ID)
```
Comment (ChrilleSigerhed) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionOvrigtKomplikationerAvSjukdomen.java:51]: Should the config.id be OVRIGT_KOMPLIKATIONER_AV_SJUKDOMEN_JSON_ID instead? This has no functional impact from what I can tell, but to stay uniformed in how we design our configs, consider making this change.

**Review by ChrilleSigerhed — CHANGES_REQUESTED** [review]
Great work with this PR, real clean code with high standards, impressive!! 

I also manually tested throught the changes made to ts-diabetes-v4 and everything works just as intended, awsome!

I spotted that on some questions we use the DELSVAR_ID as config id. From what I can tell this has no functional impact, but to keep our configs uniformed, i suggest we use the JSON_ID instead.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiAllvarligSenasteTolvManaderna.java (hunk 1 above)
Comment (mikeyOne) [reply ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiAllvarligSenasteTolvManaderna.java:56]: done

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiAterkommandeVaketSenasteTolv.java (hunk 3 above)
Comment (mikeyOne) [reply ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiAterkommandeVaketSenasteTolv.java:55]: done

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiFormagaKannaVarningstecken.java (hunk 4 above)
Comment (mikeyOne) [reply ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiFormagaKannaVarningstecken.java:55]: done

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiForstarRiskerMedHypoglykemi.java (hunk 5 above)
Comment (mikeyOne) [reply ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiForstarRiskerMedHypoglykemi.java:56]: done

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiKontrollSjukdomstillstand.java (hunk 6 above)
Comment (mikeyOne) [reply ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiKontrollSjukdomstillstand.java:55]: done

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiRegelbundnaBlodsockerkontroller.java (hunk 7 above)
Comment (mikeyOne) [reply ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiRegelbundnaBlodsockerkontroller.java:57]: done

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiVidtaAdekvataAtgarder.java (hunk 8 above)
Comment (mikeyOne) [reply ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionHypoglykemiVidtaAdekvataAtgarder.java:55]: done

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionOvrigtBorUndersokasAvSpecialist.java (hunk 9 above)
Comment (mikeyOne) [reply ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionOvrigtBorUndersokasAvSpecialist.java:50]: done

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionOvrigtKomplikationerAvSjukdomen.java (hunk 10 above)
Comment (mikeyOne) [reply ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/certificate/question/QuestionOvrigtKomplikationerAvSjukdomen.java:51]: done

**Comment by mikeyOne** [comment]
Same maneuver (DELSVAR_ID -> JSON_ID) done also in QuestionBedomningOvrigaKommentarer.java

**Review by ChrilleSigerhed — APPROVED** [review]
Awsome, approved!

**Pull request merged**